nav2_bringup/launch/slam_launch.py

```diff
@@ -44,82 +44,118 @@ def generate_launch_description():
     slam_launch_file = os.path.join(slam_toolbox_dir, 'launch', 'online_sync_launch.py')
 
     # Create our own temporary YAML files that include substitutions
-    param_substitutions = {
```

Comment:
Please readd (+ use in `ParameterFile` below). Its a little unclear to me why this was removed? 

This exists so that if you have `use_sim_time` in your param file, it can be overridden before being passed to the nodes based on the commandline argument. Its a subtle but important feature